src/ragger/utils/misc.py

```diff
@@ -47,10 +47,12 @@ def find_library_application(base_dir: Path, name: str, device: str) -> Path:
     return lib
 
 
-def find_main_application(base_dir: Path, device: str) -> Path:
+def find_main_application(base_dir: Path, build_dir: str, bin_name: str, device: str) -> Path:
     if not base_dir.is_dir():
         raise AssertionError(f"{base_dir} is not a directory")
-    app = Path(base_dir / "build" / device / "bin" / "app.elf").resolve()
+    app = Path(base_dir / build_dir / device / bin_name).resolve()
+    if not app.is_file() and device == "nanos2":
+        app = Path(base_dir / build_dir / "nanosp" / bin_name).resolve()
```

Comment:
"nanosp" value was explicitely converted to "nanos2" before in the code, as `C` SDK creates a `nanos2` build directory. Does this implies the Rust SDK use "nanosp" as directory name?
We seriously need to make a choice and stick to it on this. It's comical that we still have bugs triggering and have to add this kind of opaque conversion here and there for such a trivial issue.

Reply:
Actually, I made a mistake as Rust SDK is using `nanosplus` (derived from target name) as directory name...

Reply:
So if we call `pytest` with `--device nanosplus` it should be ok without this modification, shouldn't it ?